lib/csv/writer.rb

```diff
@@ -156,7 +156,7 @@ def quote(field)
         else
           field = String(field)  # Stringify fields
           # represent empty fields as empty quoted fields
-          if (@quote_empty and field.empty?) or @quotable_pattern.match?(field)
+          if (@quote_empty and field.empty?) or @quotable_pattern.match?(field.scrub)
```

Comment:
I don't think that this is a good approach.
If we have `\uFFFD` in `@quotable_pattern`, it matches all invalid characters.

We need to process this case carefully something like:

 ```suggestion
          if (@quote_empty and field.empty?) or (filed.valid_encoding? and @quotable_pattern.match?(field)
```

Could you create a separated pull request for this case and then we can back to this pull request?

Reply:
Thanks for your suggestion that definitely a safer approach. I think that is definitely a safe approach and suitable. I've opened #131.

Reply:
I rebased this PR with the latest mater branch.